configs/components/ruby-2.1.9.rb

```diff
@@ -5,21 +5,9 @@
   pkg.mirror "#{settings[:buildsources_url]}/ruby-#{pkg.get_version}.tar.gz"
```

Comment:
There's some shared code here and in ruby-2.4.2, from what I can find here:

https://github.com/caseywilliams/puppet-runtime/blob/agent-runtime/configs/components/ruby-2.1.9.rb#L7-L9
https://github.com/caseywilliams/puppet-runtime/blob/agent-runtime/configs/components/ruby-2.4.2.rb#L7-L9

Common patches for AIX and Windows

This part:
https://github.com/caseywilliams/puppet-runtime/blob/agent-runtime/configs/components/ruby-2.1.9.rb#L127-L130
https://github.com/caseywilliams/puppet-runtime/blob/agent-runtime/configs/components/ruby-2.4.2.rb#L128L131

This part:
https://github.com/caseywilliams/puppet-runtime/blob/agent-runtime/configs/components/ruby-2.1.9.rb#L132-L142
https://github.com/caseywilliams/puppet-runtime/blob/agent-runtime/configs/components/ruby-2.4.2.rb#L133-L144

(one has an extra zlib-devel on AIX, aside from that things are the same)

This part:
https://github.com/caseywilliams/puppet-runtime/blob/agent-runtime/configs/components/ruby-2.1.9.rb#L144-L150
https://github.com/caseywilliams/puppet-runtime/blob/agent-runtime/configs/components/ruby-2.4.2.rb#L146-L152

Most of these parts:
https://github.com/caseywilliams/puppet-runtime/blob/agent-runtime/configs/components/ruby-2.1.9.rb#L152-L275
https://github.com/caseywilliams/puppet-runtime/blob/agent-runtime/configs/components/ruby-2.4.2.rb#L171-L291

If you're interested, I have some ideas on how to clean this stuff up